store/postgres/src/deployment_store.rs

```diff
@@ -1353,7 +1354,14 @@ impl DeploymentStore {
                 // Revert the data
                 let layout = self.layout(conn, site.clone())?;
 
-                let (event, count) = layout.revert_block(conn, block)?;
+                let event: StoreEvent;
+                let count: i32;
+
+                if truncate {
+                    (event, count) = layout.truncate_tables(conn)?;
+                } else {
+                    (event, count) = layout.revert_block(conn, block)?;
+                }
```

Comment:
You can write this more idiomatically as
```rust
let (event, count) = if truncate { .. }
```

Actually, I would pull the other block from line 1373 into this, too, and write this as
```rust
let event = if truncate {
                      let event = layout.truncate_tables(conn)?;
                      deployment::set_entity_count(conn, site.as_ref(), layout.count_query.as_str())?;
                      event
                   } else {
                      let (event, count) = layout.revert_block(conn, block)?;
                      deployment::update_entity_count(conn, site.as_ref(), count)?;
                      event
                  };
```

That avoids passing around a count that means very different things for rewinding vs truncate